docs/reference/contributing/guidelines/workflow.md

```diff
@@ -94,10 +94,18 @@ Release: patch
 
 #### Functionality change
 
-Any change in the functionality, it can be adding a new feature, adding a new method or a function. Software language does not matter.
+A functionality change can be any change in the functionality, including adding a new feature, a new method or a function. Software language does not matter.
 
 A feature contribution contains a new API, capability or behavior. It does not break backward compatibility with existing APIs, capabilities or behaviors. New feature contributions are very welcome in Mbed OS. However, because they add capability to the codebase, it's easy for a new feature to introduce bugs and a support burden. The introduction of new features should also come with documentation, majority of targets support and comprehensive test coverage proving the correctness of the feature per the documentation. Feature PRs are treated cautiously, and new features require a new minor version for the codebase. Features are candidates for feature releases. 
 
+Every pull request changing or adding functionality must contain a release notes section called "Release notes" to describe the changes to users.
+
+It must contain:
+
+- A brief description of changes introduced.
+- An impact analysis: components affected, potential consequences for users and reasons for the addition or change.
```

Comment:
Query: Can we use a word other than "impact" here? It's technically imprecise language. ("Impact" implies hitting something, such as impacted teeth or a meteor impact.)